### DISCUSSION TITLE
Apple OAuth2 Setup

### DISCUSSION THREAD

**Discussion by ThePuscher** [appwrite/appwrite#3808] [Q&A]
Dear Appwrite community,

I am trying to set up sign in with apple for my iOS app.
I completed [Getting Started for Apple](https://appwrite.io/docs/getting-started-for-apple) and [Create Account Session with OAuth2](https://appwrite.io/docs/client/account?sdk=apple-default#accountCreateOAuth2Session) from the docs.
I also added Apple as a platform to my appwrite app and enabled Apple as an OAuth2 provider under Users > Settings, using the data from a key I created on my Apple developer account.

In the menu it still says: "To complete set up, add this OAuth2 redirect URI to your Apple app configuration." "http://localhost/v1/account/sessions/oauth2/callback/apple/some_key".
I do not know how to complete this step, help would be very appreciated!

Now, when trying to login, it redirects to Apple but yields the error "Invalid web redirect url.".

Kind regards,
Josh

**Comment by stnguyen90** [comment]
Hi @ThePuscher! Maybe [this](https://dev.to/appwrite/apple-sign-in-with-appwrite-2576) guide will help?

**Reply by ThePuscher** [reply]
Dear @stnguyen90, thank you for your answer!

My problem occurs during the "Create a Service ID" step, I don't know what to fill in for Domains and Subdomains on Apple, if the login is only in the app and not on a website.

For the Return URLs, appwrite gives http://localhost/v1/account/sessions/oauth2/callback/apple/..., which is not accepted by Apple because it is not https, so I am probably required to host the website publicly?

**Reply by stnguyen90** [reply]
@ThePuscher, 

> My problem occurs during the "Create a Service ID" step, I don't know what to fill in for Domains and Subdomains on Apple, if the login is only in the app and not on a website.

You should be using your Appwrite instance because the OAuth2 flow is actually between Appwrite and Apple, not your app and Apple.

> For the Return URLs, appwrite gives http://localhost/v1/account/sessions/oauth2/callback/apple/..., which is not accepted by Apple because it is not https, so I am probably required to host the website publicly?

You can access Appwrite using https://localhost (assuming you used port 443 for the HTTPS port during installation), but you'll just need to accept the self signed certificate warning.

**Comment by Shiba-Kar** [comment]
![image](https://user-images.githubusercontent.com/24454722/225272295-78981079-40c1-40fe-b2f9-202298e61097.png)

In the latest appstore connect there is no  Domains and Subdomains  fields to add. 
where can I paste the redirect URL ?

![image](https://user-images.githubusercontent.com/24454722/225273119-b7deeb83-1128-4a67-8147-98dd36f3343c.png)

I guess the above steps just gone obsolete.

**Reply by stnguyen90** [reply]
@Shiba-Kar, from what I can tell, the instructions are still valid. You should be creating the service in the [Developer portal](https://developer.apple.com/account/resources/identifiers/list/serviceId).
